common/src/model_deployment/deploy.py

```diff
@@ -234,31 +221,85 @@ def main(args):
+    except Exception as e:
+        error_msg = f"Error occured while creating endpoint - {e}"
+        logging.error(error_msg)
+        raise Exception(error_msg)
+
+    try:
+        print(f"Creating deployment {deployment}")
         ml_client.online_deployments.begin_create_or_update(deployment).wait()
     except Exception as e:
         error_msg = f"Error occured while creating deployment - {e}"
         logging.error(error_msg)
         raise Exception(error_msg)
 
-    print("Deployment done with name ", deployment.name)
+    print(f"Deployment successful. Updating endpoint to take 100% traffic for deployment {deployment_name}")
 
     # deployment to take 100% traffic
     endpoint.traffic = {deployment.name: 100}
     try:
         ml_client.begin_create_or_update(endpoint).wait()
+        endpoint = ml_client.online_endpoints.get(endpoint.name)
     except Exception as e:
         error_msg = f"Error occured while updating endpoint traffic - {e}"
         logging.error(error_msg)
         raise Exception(error_msg)
 
+    return endpoint, deployment
+
+
+def main(args):
+    """Run main function."""
+    ml_client = get_ml_client()
+    # get registered model id
+    model_info = {}
+    with open(args.registration_details) as f:
+        model_info = json.load(f)
+    model_id = model_info["id"]
+    model_name = model_info["name"]
+
+    # make sure underscores and slashes are replaced by hyphens and convert them to lower case
+    endpoint_name = re.sub('[/_ ]', '-', model_name)
+    endpoint_name = f"{endpoint_name.lower()}-{int(time.time())}"
+    endpoint_name = endpoint_name[:32].lower()
+
+    endpoint_name = args.endpoint_name if args.endpoint_name else endpoint_name
+    deployment_name = args.deployment_name if args.deployment_name else "default"
+
+    endpoint, deployment = create_endpoint_and_deployment(
+        ml_client=ml_client,
+        endpoint_name=endpoint_name,
+        deployment_name=deployment_name,
+        model_id=model_id,
+        args=args
+    )
+
+    if args.inference_payload:
+        print("Invoking inference with test payload ...")
+        try:
+            response = ml_client.online_endpoints.invoke(
+                endpoint_name=endpoint_name,
+                deployment_name=deployment_name,
+                request_file=args.inference_payload,
+            )
+            print(f"Response:\n{response}")
+        except Exception as e:
+            print(f"Invocation failed with error: {e}")
+            raise e
+
+    print("Saving deployment details ...")
+
     # write deployment details to file
     endpoint_type = "aml_online_inference"
-    endpoint_response = ml_client.online_endpoints.get(endpoint.name)
     deployment_details = {
         "endpoint_name": endpoint.name,
         "deployment_name": deployment.name,
-        "endpoint_uri": endpoint_response.__dict__["_scoring_uri"],
+        "endpoint_uri": endpoint.__dict__["_scoring_uri"],
```

Comment:
What's going on here?

Reply:
Removed fetching the endpoint details again, and sending response which we receive as part of endpoint traffic update.